src/tool.js

```diff
@@ -0,0 +1,39 @@
+exports.genForm = (inputs) => {
+    // Create a blank string to start with
+    var formStr = '';
+    try {
+        inputs.forEach((input) => {
+            // Make sure input has property `type`
+            if (input.type) {
```

Comment:
Instead of having if and switch statement you could just have an array with valid input types and one if statement to check if the values are in the array. It could save you a lot of lines!

example:
```js
let validInputs = ['text', 'password', 'label', 'select', 'reset', 'submit'];

if(validInputs.includes(input.type)){

}
```

Reply:
This comment seems to address the same concern as your comment below, so I'm only responding to one.

I understand the approach you're trying to take here, and I appreciate opportunities to eliminate lines of code.

That said, I'd prefer to keep the `switch` expression over your pattern for the following reasons:

- Some HTML form elements are not `input` tags and are not self-closing (`select` and `label` are two examples)
- Not all HTML attributes are supported by each type of input or form element
- Future scalability might include things like adding an `option` property to the `select` type with an array of objects, which doesn't work with this one-size-fits-all approach

To elaborate on each point: `select` and `label` elements both require closing tags (along with `textarea` and a handful of others not currently supported by this tool. A pattern like

```
if(validInputs.includes(input.type)){
  returnString=`<input type="${input.type}" ... />`;
}
```

can't accommodate something like `select`.

Not all of the HTML form elements take the same attributes. If the end goal is to produce valid, semantic, syntactically-compliant HTML forms dynamically, there is no benefit to passing, for instance, a `for` attribute to anything except `label` elements. To account for this, we'd need to provide some conditionality anyway, and we wind up with a stack of `if...else` statements. For all the different combinations of supported attributes, we'd need quite a few of them, which isn't great:

>In JavaScript, `if` statements are generally faster than `switch` statements when there are just _one or two conditions_ to be evaluated. When there are more than two conditions, and the conditions are simple (not ranges), the `switch` statement tends to be faster.

(Read more about efficiency in Javascript [here](http://archive.oreilly.com/pub/a/server-administration/excerpts/even-faster-websites/writing-efficient-javascript.html)).

Finally, the pattern above can't accommodate nested elements, like `option`s inside a `select`, or a bunch of radio buttons inside a `fieldset`.

In this case, I believe the `switch` expression to be a much simpler and more elegant solution, despite adding weight to the code.

Reply:
I agree with you on that! So probably you could combine both options and use the switch statement according to Html tag and not the input!